drivers/crypto/crypto_mtls_shim.c

```diff
@@ -44,12 +48,102 @@ struct mtls_shim_session mtls_sessions[CRYPTO_MAX_SESSION];
 #error "You need to define MBEDTLS_MEMORY_BUFFER_ALLOC_C"
 #endif /* MBEDTLS_MEMORY_BUFFER_ALLOC_C */
 
+int mtls_ecb_encrypt(struct cipher_ctx *ctx, struct cipher_pkt *pkt)
+{
+	int ret;
+	mbedtls_aes_context *ecb_ctx =
+		&((struct mtls_shim_session *)ctx->drv_sessn_state)->mtls_aes;
```

Comment:
Maybe you could add a macro to do this: `((struct mtls_shim_session *)ctx->drv_sessn_state)` since this is done in a few places.

Reply:
something like this?
```
#define get_mtls_ctx(c, m) &((struct mtls_shim_session *)c->drv_sessn_state)->mtls_ ## m

mbedtls_aes_context *ctx = get_mtls_ctx(ctx, aes);
```

Reply:
Yes, that looks better (for me).

Reply:
Done.